kraft-jennifer-ndvi-automation.py

```diff
@@ -0,0 +1,698 @@
+    # Clip NDVI layer
+    ndvi_crop = ndvi_xr.rio.clip(crop_json)
+    
+    # Apply cloud mask to NDVI
+    ndvi_crop = ndvi_crop.where(~cl_mask_crop.isin(vals))
+
+    return ndvi_crop
+
+
+# In[6]:
+
+
+# Get a list of each directory
+path = os.path.join("ndvi-automation", "sites")
+
+# Get a list of both site directories (We will talk more about automation next week)
+sites = glob(path + "/*/")
+sites
+
+# Get the site name
+site_name = os.path.basename(os.path.normpath(sites[0]))
+site_name
+
+# Open up the shapefile for clipping your landsat data to the study area
+vector_dir = os.path.join(sites[0],
+                          "vector")
+
+# Open crop boundary
+site_boundary_path = os.path.join(vector_dir,  site_name + "-crop.shp")
+crop_bound = gpd.read_file(site_boundary_path)
+crop_bound.plot()
+plt.show()
+
+landsat_dir = os.path.join(sites[0],
+                           "landsat-crop")
+# This is the crop folder containing all of the .tif files
+landsat_dir
+
+#Create lists of subdirectories
+landsat_dirs = sorted(glob(os.path.join(landsat_dir, "LC08*")))
+landsat_dirs
+
+# Select just a single directory and grab bands 4-5 from the directory
+adir = landsat_dirs[4]
+
+# Open bands
+band_paths = sorted(glob(os.path.join(adir, "*band*[4-5].tif")))
+band_paths
+
+# Open & clip the data - a single band using rioxarray
+band = rxr.open_rasterio(band_paths[0], masked=True).rio.clip(crop_bound.geometry,
+                                                              from_disk=True).squeeze()
+
+# Specify the valid range of values for landsat
+valid_range = (0, 10000)
+
+if valid_range:
+    mask = ((band < valid_range[0]) | (band > valid_range[1]))
+    band = band.where(~xr.where(mask, True, False))
+
+band
+
+# Open and clean a single band using a function
+band_4 = open_clean_bands(band_path=band_paths[0],
+                          crop_extent=crop_bound,
+                          # The range of valid values for landsat can include negative values
+                          # for this week let's stick with 0-10000
+                          valid_range=(0, 10000))
+
+# %%timeit
+
+# Open and clean a single band
+band_4 = open_clean_bands(band_path=band_paths[0],
+                          crop_extent=crop_bound,
+                          # The range of valid values for landsat can include negative values
+                          # for this week let's stick with 0-10000
+                          valid_range=(0, 10000))
+
+# Open and clean a single band
+band_5 = open_clean_bands(band_path=band_paths[1],
+                          crop_extent=crop_bound,
+                          # The range of valid values for landsat can include negative values
+                          # for this week let's stick with 0-10000
+                          valid_range=(0, 10000))
+
+#  Then  calculate NDVI
+ndvi = ((band_5-band_4) / (band_5+band_4))
+
+ep.plot_bands(ndvi,
+              cmap='PiYG',
+              vmin=-1)
+plt.show()
```

Comment:
This also seems left over from working through the code, and isn't necessary to final product.